### ISSUE TITLE
Is this project maintained?

### ISSUE THREAD

**Issue by Brian-McBride** [ExpressGateway/express-gateway#1011]
The last release: **Sep 22, 2019**

Anyone know if this is maintained or maybe there is another project that picked up where this one left off?

**Comment by fourlincoln10** [comment]
Wondering the same thing myself

**Comment by XVincentX** [comment]
Hey,

as I have already clarified in other occasions, the project is not actively maintained anymore.

I have put a significant amount of effort into this, but unfortunately:

* I haven't been able to find external contributors
* Not a single big company using the project (including Dell, BBVA and FIFA by the way) have been either willing to give a testimonial or fund the project somehow.

Sometimes it's mature to realize that a project is a failure — and ultimately (and unfortunately), Express Gateway was one of these.

**Comment by fourlincoln10** [comment]
Completely understand Vincenzo. Thank you for the work you've done!

**Comment by marcosalpereira** [comment]
I hope that some company will review its position and help the project

**Comment by XVincentX** [comment]
Nothing as changed so far @marcosalpereira

**Comment by tagkiller** [comment]
It is sad, if I had a bit more time on my hand I'd gladly have offered my help as a contributor. Anyway thanks for the great work @XVincentX !

**Comment by akira28** [comment]
thanks for your work @XVincentX ! Sorry to ear it was a failure :(
Right now we are planning to build an api gateway for our services, do you have any suggestion which project should we look at? Thank you very much

**Comment by XVincentX** [comment]
There's a number of mature project you can take a look at. If you want to play safe, Kong, Tyk and WSO2 are all good solution.

Cheers!

**Comment by adhisimon** [comment]
Is there any recomendation for an alternative? We need API Gateway based on node.js (especially express.js) like EG.

**Comment by scampiuk** [comment]
Always sad to hear, but a sensible approach.  If you're not actively maintaing, would you still be reviewing PRs?

May I recommend that you mention this on the website also. I've spent a day or so playing around, just to run into issues I know now most likely won't be fixed :(

Shame, this did exactly what we needed :(

**Comment by papakai** [comment]
You should add this information at the top of the README to provide that information immediately to the public.

**Comment by XVincentX** [comment]
> You should add this information at the top of the README to provide that information immediately to the public.

PR welcomed :)

**Comment by papakai** [comment]
voilà!

**Comment by fvastu** [comment]
> There's a number of mature project you can take a look at. If you want to play safe, Kong, Tyk and WSO2 are all good solution.
> 
> Cheers!

To me, the project was really good and I hope it will gain more attention and support in the future. However, as adhisimon said, can you suggest a good solution that integrates with the current express ecosystem? ExpressGateway was the perfect solution (Free, Integrated with express, easy to use..).

**Comment by Brian-McBride** [comment]
> > There's a number of mature project you can take a look at. If you want to play safe, Kong, Tyk and WSO2 are all good solution.
> > Cheers!
> 
> To me, the project was really good and I hope it will gain more attention and support in the future. However, as adhisimon said, can you suggest a good solution that integrates with the current express ecosystem? ExpressGateway was the perfect solution (Free, Integrated with express, easy to use..).

It has a price tag, but Apigee offers extensibility via NodeJS following the Express pattern.. That is the only one I can think of off the top of my head. 

If you are doing microservices and not using zero scale (your nodes are always on) then check out Molecular.js Is a little lacking in the typescript support, but it has some cool ideas. If I were to run my own on-prem service cluster, I would seriously consider picking it up. It has a gateway that can be used to provide a RESTful layer.

If your are deploying into a cloud provider, you might just try and use theor PaaS services.

**Comment by gyj1278** [comment]
我正在搞express的API网关，结果不维护了，不知道该用啥了

**Comment by adhisimon** [comment]
Just for information, in the end, I use krakend (written in golang) as express-gateway replacement.

**Comment by codezilla91** [comment]
Is this the official Express Gateway that will no longer be maintained? I do not see any public notice on the official website, can i still use it or rather look at alternatives?

**Comment by MiquelMarinoColvin** [comment]
Thanks for your effort. We were currently using this marvelous framework to build an API gateway.  Unfortunately we will change to other framework given the circunstancies 😢 
Thanks to @XVincentX and the rest of the community for your incredible effort and great idea 😄

**Comment by cingaldi** [comment]
> Hey,
> 
> as I have already clarified in other occasions, the project is not actively maintained anymore.
> 
> I have put a significant amount of effort into this, but unfortunately:
> 
> * I haven't been able to find external contributors
> * Not a single big company using the project (including Dell, BBVA and FIFA by the way) have been either willing to give a testimonial or fund the project somehow.
> 
> Sometimes it's mature to realize that a project is a failure — and ultimately (and unfortunately), Express Gateway was one of these.

late message but let me say it's very sad to know that this promising project has come to an end, thanks for your effort!

**Comment by remvee** [comment]
Hi all,

First of all thank you to @XVincentX and the rest of the community for building EG.  We've been using it for a project and have been pretty happy with it because of the ease of extending EG.  We came a little late to the EG party and maintenance was dropped just as we finished our first EG-based project.  :-(  So.. we decide to fork and make a "lite" edition dropping all the stuff we don't need, in an effort to reduce the amount of dependencies thus making it easier to maintain.

If anybody is interested: https://github.com/jomco/express-gateway-lite

Cheers,
R.
